### PR TITLE
algorithms: expose Proj parameter in hpx::min/max/minmax_element CPOs

### PR DESCRIPTION
**Pull request by aneek22112007-tech**
## Summary
This PR exposes the `Proj` (projection) parameter in the public Customization Point Objects (CPOs) for `hpx::min_element`, `hpx::max_element`, and `hpx::minmax_element`. 

Previously, while the internal `detail` layer supported projections, the public parallel CPO overloads hardcoded `hpx::identity_v`. This change brings these algorithms into alignment with the `std::ranges` design and ensures consistency with the existing `hpx::ranges` container algorithms.

### CPO Updates
- Updated all six `tag_fallback_invoke` overloads for each algorithm in `libs/core/algorithms/include/hpx/parallel/algorithms/minmax.hpp`.
- Added a `Proj` template parameter (defaulting to `hpx::identity`) to support the standard projection API across all execution policies (including `seq`, `par`, and `async` variations).
- Integrated `hpx::parallel::traits::is_projected_v` and `hpx::parallel::traits::is_indirect_callable_v` constraints to ensure strict type checking and improved error diagnostics at the CPO boundary.

### Type Deduction Fix
- Resolved a regression in `element_type` deduction within `sequential_minmax_element_ind`. 
- The previous deduction logic using `std::iter_value_t<FwdIter>&` was insufficient for the parallel reduction phase, where `FwdIter` refers to an iterator over partition results rather than the base elements. 
- Restored the `decltype(HPX_INVOKE(proj, *it))` pattern to correctly resolve the proxy value type when merging partition results.

### Testing
- Added three new regression test files: `min_element_proj.cpp`, `max_element_proj.cpp`, and `minmax_element_proj.cpp`.
- Test coverage includes:
    - Projecting member variables from custom structs/pairs.
    - Value transformations (e.g., negating values to invert min/max logic).
    - Verification of projection forwarding in `seq(task)` async policies.
    - Edge cases (empty ranges, single-element ranges, and all-equal elements).
- Confirmed that all existing `min_element`, `max_element`, and `minmax_element` unit tests pass.

## Checklist

- [x] I have added a new feature and have added tests to go along with it.
- [x] I have fixed a bug and have added a regression test.
- [x] I have added a test using random numbers; it uses a seed and the generated numbers are valid.

### CODE REVIEW

File: libs/core/algorithms/include/hpx/parallel/algorithms/minmax.hpp [hunk 1]
```diff
@@ -890,37 +891,53 @@ namespace hpx {
...
+                    hpx::execution::sequenced_policy, F,
+                    hpx::parallel::traits::projected<Proj, FwdIter>,
+                    hpx::parallel::traits::projected<Proj, FwdIter>
+                >
+            )
         // clang-format on
-        friend FwdIter tag_fallback_invoke(
-            hpx::min_element_t, FwdIter first, FwdIter last, F f = F())
+        friend FwdIter tag_fallback_invoke(hpx::min_element_t, FwdIter first,
+            FwdIter last, F f = F(), Proj proj = Proj())
```
Comment (hkaiser) [inline libs/core/algorithms/include/hpx/parallel/algorithms/minmax.hpp]: As mentioned several times before. I'd like to keep the HPX APIs as close to the standard as possible. `std::min_element` and friends don't expose a projection (https://eel.is/c++draft/alg.min.max), thus `hpx::min_element` and friends shouldn't do so either.

File: libs/core/algorithms/include/hpx/parallel/algorithms/is_partitioned.hpp [hunk 2]
```diff
@@ -255,33 +255,45 @@ namespace hpx {
...
+                hpx::parallel::traits::is_indirect_callable_v<
+                    hpx::execution::sequenced_policy, Pred,
+                    hpx::parallel::traits::projected<Proj, FwdIter>
+                >
             )
         // clang-format on
-        friend bool tag_fallback_invoke(
-            hpx::is_partitioned_t, FwdIter first, FwdIter last, Pred pred)
+        friend bool tag_fallback_invoke(hpx::is_partitioned_t, FwdIter first,
+            FwdIter last, Pred pred, Proj proj = Proj())
```
Comment (hkaiser) [inline libs/core/algorithms/include/hpx/parallel/algorithms/is_partitioned.hpp]: Same here. Please do not change the public HPX of the HPX algorithms in `namespace hpx` (except if they do not conform to the equivalent APIs in `namespace std`).

File: libs/core/algorithms/include/hpx/parallel/algorithms/minmax.hpp (hunk 1 above)
Comment (aneek22112007-tech) [reply libs/core/algorithms/include/hpx/parallel/algorithms/minmax.hpp]: Thank you for the feedback. I understand the importance of keeping the hpx:: namespace strictly aligned with the C++ standard. I have reverted the changes to hpx::min_element, hpx::max_element, and hpx::minmax_element to remove the projection parameter, ensuring they match the std:: API exactly. The updated commit now only focuses on adding test coverage for the hpx::ranges overloads.

Comment (aneek22112007-tech) [reply libs/core/algorithms/include/hpx/parallel/algorithms/minmax.hpp]: I apologize for the oversight. I have reverted the changes to hpx::is_partitioned and hpx::is_sorted to restore their standard-compliant signatures. All non-standard projection parameters have been removed from the hpx:: namespace CPOs in this PR.

**Review by hkaiser — APPROVED** [review]
LGTM, thanks!
